src/mem/mem_ctrl.cc

```diff
@@ -187,24 +198,120 @@ MemCtrl::writeQueueFull(unsigned int neededEntries) const
     return  wrsize_new > writeBufferSize;
 }
 
+/*bool
+MemCtrl::writeQueueFull(unsigned int neededEntries) const
+{
+    DPRINTF(MemCtrl,
+            "Write queue limit %d, current size %d, entries needed %d\n",
+            writeBufferSize, totalWriteQueueSize, neededEntries);
+
+    auto wrsize_new = (totalWriteQueueSize + neededEntries);
+    return  wrsize_new > writeBufferSize;
+}
+
+bool
+MemCtrl::writeQueueFull(unsigned int neededEntries) const
+{
+    DPRINTF(MemCtrl,
+            "Write queue limit %d, current size %d, entries needed %d\n",
+            writeBufferSize, totalWriteQueueSize, neededEntries);
+
+    auto wrsize_new = (totalWriteQueueSize + neededEntries);
+    return  wrsize_new > writeBufferSize;
+}*/
+
+bool
+MemCtrl::nvmWriteQueueFull(unsigned int neededEntries) const
+{
+    auto size = (nvmWriteQueueSize + neededEntries);
+    // random size to compare with for now
+    //return  size_new > 64;
+    return  size > 64;
```

Comment:
64 should be a maxNvmWriteQueueSize parameter

Reply:
Should these vars (e.g. maxNvmWriteQueueSize) be 'Param' or regular const vars?

Reply:
I'd go ahead and make them params... You'll probably want this at some point anyway.